File: api/models/page.go
package models

// Generic type for returning paginated results.
type Page[T any] struct {
	Items       []T   `json:"items"`
	TotalPages  int64 `json:"totalPages"`
	Page        int64 `json:"page"`
	PageSize    int64 `json:"pageSize"`
	HasNextPage bool  `json:"hasNextPage"`
}

File: api/Models/Page.cs
namespace pastemyst.Models;

public class Page<T>
{
    public List<T> Items { get; set; } = new();

    public int TotalPages { get; set; }

    public int CurrentPage { get; set; }

    public int PageSize { get; set; }

    public bool HasNextPage { get; set; }
}
